core/app/models/spree/order.rb

```diff
@@ -173,6 +173,10 @@ def amount
       line_items.inject(0.0) { |sum, li| sum + li.amount }
     end
 
+    def promotionable_amount
+      line_items.promotionable.map(&:amount).sum
```

Comment:
```ruby
line_items.promotionable.pluck(:amount).sum
```

Reply:
`amount` is money field. You cannot pluck it